mlrun/datastore/sources.py

```diff
@@ -361,13 +389,12 @@ def to_step(
     ):
         import storey
 
-        attributes = self.attributes or {}
+        attributes = deepcopy(self.attributes)
```

Comment:
Isn't a regular copy enough?

Reply:
no, if you have "in" operator it can function as a nested dict.